### PR TITLE
UPSTREAM: <carry>: openshift: Move machineSpec from azureprovider.k8s.io/v1alpha1 into azureproviderconfig.openshift.io/v1beta1

### PR DESCRIPTION
**Pull request by enxebre**
Move machineSpec from azureprovider.k8s.io/v1alpha1 into azureproviderconfig.openshift.io/v1beta1

### CODE REVIEW

File: pkg/apis/azureprovider/v1beta1/doc.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1beta1 contains API Schema definitions for the azureprovider v1beta1 API group
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen=package,register
+// +k8s:conversion-gen=sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider
+// +k8s:defaulter-gen=TypeMeta
+// +groupName=azureprovider.k8s.io
```
Comment (frobware) [inline pkg/apis/azureprovider/v1beta1/doc.go]: Should this be: `azureproviderconfig.openshift.io`?

Comment (enxebre) [reply pkg/apis/azureprovider/v1beta1/doc.go]: thanks!

**Comment by ingvagabund** [comment]
Azure actuator is actually using `yaml.Unmarshal`, which ignores machine `ApiVersion` field. So, it's safe to merge this without any effect on the installer.

**Pull request merged**